code/WorkInProgress/cloner_defects.dm

```diff
@@ -64,6 +64,12 @@
 					// lazylistassoc when (wici)
 					sublist[defect_type] = initial(defect_type:weight) // we can use initial to get initial values of a type without an instance
 
+	disposing()
+		src.owner = null
+		for (var/datum/cloner_defect/defect as anything in src.active_cloner_defects)
+			defect.dispose()
```

Comment:
Any particular reason why dispose instead of qdel?

Also you should null the active_cloner_defects list here too.

Does owner have no reference to src here? Like onwer.cloner_defects or something?

Reply:
I disposed because when I was checking human disposing it called dispose on things, not sure which was best so I just went with dispose().

Reply:
active_cloner_defects should get nulled at the end of the loop?

Reply:
`qdel` is almost always preferred over manually calling dispose unless you have a reason to do otherwise